tests/foreman/api/test_computeresource_gce.py

```diff
@@ -117,6 +118,11 @@ def gce_hostgroup(
     return hgroup
 
 
+@pytest.fixture(scope='module', autouse=True)
+def download_cert():
+    download_gce_cert()
```

Comment:
Nothing deletes the downloaded cert, so this and the other fixtures will be downloading the cert multiple times on a target satellite instance.  I count 4 calls total to this method, each one just writing the same cert to the same target file.

I think you can just define a session scoped fixture in `pytest_fixtures/sys_fixtures` that checks if the cert is set in settings and calls this helper if it is.  Not `autouse`, so that the download is only done if tests that require it have been collected and are executing.

The fixture could return a boolean indicating whether the cert was downloaded, and all the fixtures/tests that depend on the cert can use that fixture.  Since the download fixture would be session scoped, it would only execute once even though it has multiple test dependencies.

It would also give an easy way for these GCE tests to call pytest.skip if the cert isn't available.